client.go

```diff
@@ -49,38 +50,78 @@ type MetricOutput struct {
 	RetriesFailed uint64 `json:"retries_failed"`
 }
 
+type PluginStorage struct {
+	Status struct {
+		Overlimit bool   `json:"overlimit"`
+		MemSize   string `json:"mem_size"`
+		MemLimit  string `json:"mem_limit"`
+	} `json:"status"`
+
+	Chunks struct {
+		Total    int    `json:"total"`
+		Up       int    `json:"up"`
+		Down     int    `json:"down"`
+		Busy     int    `json:"busy"`
+		BusySize string `json:"busy_size"`
+	} `json:"chunks"`
+}
+
+type StorageMetrics struct {
+	StorageLayer struct {
+		Chunks struct {
+			TotalChunks  int `json:"total_chunks"`
+			MemChunks    int `json:"mem_chunks"`
+			FsChunks     int `json:"fs_chunks"`
+			FsChunksUp   int `json:"fs_chunks_up"`
+			FsChunksDown int `json:"fs_chunks_down"`
+		} `json:"chunks"`
+	} `json:"storage_layer"`
+
+	InputChunks map[string]PluginStorage `json:"input_chunks"`
+}
+
 func (c *Client) BuildInfo(ctx context.Context) (BuildInfo, error) {
 	var info BuildInfo
-	return info, c.fetchJSON(ctx, "/", &info)
+	return info, c.fetchJSON(ctx, "/", 0, &info)
 }
 
 func (c *Client) UpTime(ctx context.Context) (UpTime, error) {
 	var up UpTime
-	return up, c.fetchJSON(ctx, "/api/v1/uptime", &up)
+	return up, c.fetchJSON(ctx, "/api/v1/uptime", 0, &up)
 }
 
 func (c *Client) Metrics(ctx context.Context) (Metrics, error) {
 	var mm Metrics
-	return mm, c.fetchJSON(ctx, "/api/v1/metrics", &mm)
+	return mm, c.fetchJSON(ctx, "/api/v1/metrics", 0, &mm)
+}
+
+func (c *Client) StorageMetrics(ctx context.Context) (StorageMetrics, error) {
+	var mm StorageMetrics
+	return mm, c.fetchJSON(ctx, "/api/v1/storage", 3, &mm)
 }
 
-func (c *Client) fetchJSON(ctx context.Context, endpoint string, ptr interface{}) error {
+const DefaultHTTPRetryBackoff = 1 * time.Second
+
+func (c *Client) fetchJSON(ctx context.Context, endpoint string, retries int, ptr interface{}) error {
 	req, err := http.NewRequestWithContext(ctx, http.MethodGet, c.BaseURL+endpoint, nil)
 	if err != nil {
 		return fmt.Errorf("could not create request: %w", err)
 	}
-
-	resp, err := c.HTTPClient.Do(req)
-	if err != nil {
-		return fmt.Errorf("could not do request: %w", err)
+	var resp *http.Response
+	for retries >= 0 {
+		resp, err = c.HTTPClient.Do(req)
+		if err == nil && resp.StatusCode != http.StatusNotFound {
+			break
+		}
+		retries -= 1
+		time.Sleep(DefaultHTTPRetryBackoff)
```

Comment:
I always prefer to use `time.NewTicker(d).C` with a `for select` so you can stop also on `ctx.Done()`.